pkg/flowaggregator/flowaggregator.go

```diff
@@ -724,3 +818,130 @@ func (fa *flowAggregator) createInfoElementForTemplateSet(ieName string, enterpr
 	}
 	return ie, nil
 }
+
+// updateIfConfigChanges has a watcher to detect if modification happens. Then check whether we should update clickhouse or flow-collector at runtime
+func (fa *flowAggregator) updateIfConfigChanges(stopCh <-chan struct{}) {
+	if fa.configData == nil {
+		var err error
+		fa.configData, err = ioutil.ReadFile(fa.configFile)
```

Comment:
you need to read the file _after_ starting the watcher, otherwise you can miss changes

see https://github.com/antrea-io/antrea/pull/1427/files for an old example I wrote

Reply:
Do we have the same issue (watcher will stop watching after certain time) for the fsnotify.watcher?

Reply:
It looks like fsnotify.watcher won't be disconnected after certain period. So I moved it to option.go and also removed the  wait.NonSlidingUntil, which was used to prevent watcher from stopping.